orangecontrib/text/widgets/tests/test_owannotator.py

```diff
@@ -197,6 +197,15 @@ def test_attr_label_metas(self):
         simulate.combobox_activate_item(self.widget.controls.attr_label,
                                         self.corpus.domain[-1].name)
 
+    def test_attr_heuristics(self):
+        widget = self.create_widget(OWAnnotator)
+        c = preprocess(Corpus.from_file("slo-opinion-corpus.tab"))
+        self.send_signal(widget.Inputs.corpus, c[:20])
+        self.wait_until_finished()
```

Comment:
When you create a new widget within a test you need to pass it to the wait_until_finished:
`self.wait_until_finished(widget=widget)`

Beside that, the wait_until_finished has nothing to do with the attributes. It waits for `run` to finish, i.e. `on_done` to happen. 
However, it should be called (wait_until_finished) anyway not to mess up the other tests.